File: Week_6/problem_4-Print_even_indices.py
# https://codeforces.com/group/MWSDmqGsZm/contest/223339/problem/F

def even_num_fun(ar, n, i):
    if n < i + 1:
        return
    
    # Check if the index is even
    if n % 2 != 0:
        print(ar[n - 1], end=" ")

    even_num_fun(ar, n - 1, i)


# taking inputs
n = int(input())
ar = list(map(int, input().split()))
i = 0

even_num_fun(ar, n, i)

File: Week_6/problem_4-Print_even_indices.c
// https://codeforces.com/group/MWSDmqGsZm/contest/223339/problem/F

#include <stdio.h>
#include <limits.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

void even_num_fun(int ar[], int n, int i)
{
    if (n < i + 1)
        return;

    // Check if the index is even
    if (n % 2 != 0)
    {
        printf("%d ", ar[n - 1]);
    }

    even_num_fun(ar, n - 1, i);
}

int main()
{
    int n;
    scanf("%d", &n);
    int ar[n];

    for (int i = 0; i < n; i++)
    {
        scanf("%d", &ar[i]);
    }

    int i = 0;
    even_num_fun(ar, n, i);

    return 0;
}
